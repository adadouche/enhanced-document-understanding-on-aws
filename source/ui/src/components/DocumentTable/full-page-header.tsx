/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import React from 'react';
import { BreadcrumbGroup, Button, Header, HeaderProps, SpaceBetween } from '@cloudscape-design/components';
import { useNavigate } from 'react-router-dom';
import { InfoLink } from './info-link';

interface CasePageHeaderProps extends HeaderProps {
    title?: string;
    createButtonText?: string;
    extraActions?: React.ReactNode;
    selectedItemsCount: number;
    refreshFunction: () => void;
    onInfoLinkClick?: () => void;
}

interface DocResultsPageHeaderProps extends HeaderProps {
    title?: string;
    breadCrumbItems: {
        selectedCaseName: string;
        selectedDocumentName: string;
    };
    extraActions?: React.ReactNode;
    onInfoLinkClick?: () => void;
}

export function CasePageHeader({
    title = 'Cases',
    createButtonText = 'Create case',
    extraActions = null,
    selectedItemsCount,
    refreshFunction,
    onInfoLinkClick,
    ...props
}: CasePageHeaderProps) {
    const navigate = useNavigate();

    return (
        <Header
            variant="awsui-h1-sticky"
            info={onInfoLinkClick && <InfoLink onFollow={onInfoLinkClick} ariaLabel={`Information about ${title}`} />}
            actions={
                <SpaceBetween size="xs" direction="horizontal">
                    {extraActions}
                    <Button
                        data-testid="refresh-cases-button"
                        iconName="refresh"
                        variant="icon"
                        ariaLabel="Refresh"
                        onClick={() => refreshFunction()}
                    >
                        Refresh
                    </Button>
                    <Button data-testid="create-case-button" variant="primary" onClick={() => navigate(`/createCase`)}>
                        {createButtonText}
                    </Button>
                </SpaceBetween>
            }
            {...props}
        >
            {title}
        </Header>
    );
}

export function DocResultsPageHeader({
    title = 'Document Analysis Results',
    extraActions = null,
    onInfoLinkClick,
    breadCrumbItems,
    ...props
}: DocResultsPageHeaderProps) {
    return (
        <SpaceBetween size="m">
            <BreadcrumbGroup
                items={[
                    { text: breadCrumbItems.selectedCaseName, href: '/' },
                    { text: breadCrumbItems.selectedDocumentName, href: '#' }
                ]}
                ariaLabel="Breadcrumbs"
            />
            <Header
                variant="awsui-h1-sticky"
                info={
                    onInfoLinkClick && <InfoLink onFollow={onInfoLinkClick} ariaLabel={`Information about ${title}`} />
                }
                actions={
                    <SpaceBetween size="xs" direction="horizontal">
                        {extraActions}
                    </SpaceBetween>
                }
                {...props}
            >
                {title}
            </Header>
        </SpaceBetween>
    );
}
