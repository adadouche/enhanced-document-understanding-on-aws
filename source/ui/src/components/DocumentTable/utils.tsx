/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import { useEffect, useState } from 'react';

export const EMPTY_PANEL_CONTENT = {
    header: 'No case selected',
    body: 'Select a case to see its documents.'
};

export const useSplitPanel = (selectedItems: any) => {
    const [splitPanelSize, setSplitPanelSize] = useState(500);
    const [splitPanelOpen, setSplitPanelOpen] = useState(false);
    const [hasManuallyClosedOnce, setHasManuallyClosedOnce] = useState(false);

    const onSplitPanelResize = ({ detail: { size } }: any) => {
        setSplitPanelSize(size);
    };

    const onSplitPanelToggle = ({ detail: { open } }: any) => {
        setSplitPanelOpen(open);

        if (!open) {
            setHasManuallyClosedOnce(true);
        }
    };

    useEffect(() => {
        if (selectedItems.length && !hasManuallyClosedOnce) {
            setSplitPanelOpen(true);
        }
    }, [selectedItems.length, hasManuallyClosedOnce]);

    return {
        splitPanelOpen,
        onSplitPanelToggle,
        splitPanelSize,
        onSplitPanelResize
    };
};
