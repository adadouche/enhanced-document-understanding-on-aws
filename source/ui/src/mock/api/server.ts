// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { handlers } from "./handler";
import { setupServer } from "msw/node";
export const server = setupServer(...handlers);