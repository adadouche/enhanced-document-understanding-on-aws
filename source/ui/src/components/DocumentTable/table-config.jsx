/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import { CollectionPreferences, StatusIndicator } from '@cloudscape-design/components';

import { CaseStatusDisplayTextMapping } from "../../utils/constants";

export const CasePreferences = (props) => (
    <Preferences
        pageSizeOptions={[
            { value: 10, label: '10 cases' },
            { value: 30, label: '30 cases' },
            { value: 50, label: '50 cases' }
        ]}
        visibleContentOptions={[
            {
                label: 'Main case properties',
                options: [
                    { id: 'name', label: 'Case ID', editable: false },
                    { id: 'dateCreated', label: 'Creation Date' }
                ]
            }
        ]}
        {...props}
    />
);

const CaseStatusDisplayIndicatorMapping = {
    'initiate': 'in-progress', // displays 3 dots
    'in-process': 'loading', // displays a spinner'
    'failure': 'error',
    'success': 'success'
};

export const COLUMN_DEFINITIONS_MAIN = [
    {
        id: 'name',
        header: 'Case Name',
        cell: (item) => item.name
    },
    {
        id: 'caseId',
        header: 'Case ID',
        cell: (item) => item.caseId
    },
    {
        id: 'dateCreated',
        header: 'Creation Date',
        cell: (item) => item.dateCreated
    },
    {
        id: 'numberOfDocuments',
        header: 'Number of documents',
        cell: (item) => item.caseDocuments.length
    },
    {
        id: 'caseStatus',
        header: 'Status',
        cell: (item) => (
            <>
                <StatusIndicator type={CaseStatusDisplayIndicatorMapping[item.status]}>
                    {' '}
                    {CaseStatusDisplayTextMapping[item.status]}{' '}
                </StatusIndicator>
            </>
        )
    }
];

export const DEFAULT_PREFERENCES = {
    pageSize: 30,
    visibleContent: ['name', 'dateCreated'],
    wrapLines: false,
    stripedRows: false,
    contentDensity: 'comfortable'
};

const PAGE_SIZE_OPTIONS = [
    { value: 10, label: '10 Cases' },
    { value: 30, label: '30 Cases' },
    { value: 50, label: '50 Cases' }
];

export const Preferences = ({ preferences, setPreferences, disabled, pageSizeOptions = PAGE_SIZE_OPTIONS }) => (
    <CollectionPreferences
        title="Preferences"
        confirmLabel="Confirm"
        cancelLabel="Cancel"
        disabled={disabled}
        preferences={preferences}
        onConfirm={({ detail }) => setPreferences(detail)}
        pageSizePreference={{
            title: 'Page size',
            options: pageSizeOptions
        }}
        wrapLinesPreference={{
            label: 'Wrap lines',
            description: 'Check to see all the text and wrap the lines'
        }}
        stripedRowsPreference={{
            label: 'Striped rows',
            description: 'Check to add alternating shaded rows'
        }}
        contentDensityPreference={{
            label: 'Compact mode',
            description: 'Check to display table content in a denser, more compact mode'
        }}
    />
);
