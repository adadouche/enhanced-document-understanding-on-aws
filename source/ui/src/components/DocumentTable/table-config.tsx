/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import { CollectionPreferences, StatusIndicator } from '@cloudscape-design/components';

import { CaseStatusDisplayTextMapping } from '../../utils/constants';

export const CasePreferences = (props: any) => (
    <Preferences
        pageSizeOptions={PAGE_SIZE_OPTIONS}
        visibleContentOptions={[
            {
                label: 'Main case properties',
                options: [
                    { id: 'name', label: 'Case ID', editable: false },
                    { id: 'dateCreated', label: 'Creation Date' }
                ]
            }
        ]}
        {...props}
    />
);

const CaseStatusDisplayIndicatorMapping: any = {
    'initiate': 'in-progress', // displays 3 dots
    'in-process': 'loading', // displays a spinner'
    'failure': 'error',
    'success': 'success'
};

export const COLUMN_DEFINITIONS_MAIN = [
    {
        id: 'name',
        header: 'Case Name',
        cell: (item: any) => item.name
    },
    {
        id: 'caseId',
        header: 'Case ID',
        cell: (item: any) => item.caseId
    },
    {
        id: 'dateCreated',
        header: 'Creation Date',
        cell: (item: any) => item.dateCreated
    },
    {
        id: 'numberOfDocuments',
        header: 'Number of documents',
        cell: (item: any) => item.docCount
    },
    {
        id: 'backendUploadEnabled',
        header: 'Backend Upload Enabled',
        cell: (item: any) => item.enableBackendUpload
    },
    {
        id: 'caseStatus',
        header: 'Status',
        cell: (item: any) => (
            <>
                <StatusIndicator type={CaseStatusDisplayIndicatorMapping[item.status]}>
                    {' '}
                    {CaseStatusDisplayTextMapping[item.status]}{' '}
                </StatusIndicator>
            </>
        )
    }
];

export const DEFAULT_PREFERENCES = {
    pageSize: 20,
    visibleContent: ['name', 'dateCreated'],
    wrapLines: false,
    stripedRows: false,
    contentDensity: 'comfortable'
};

const PAGE_SIZE_OPTIONS = [
    { value: 10, label: '10 Cases' },
    { value: 20, label: '20 Cases' },
    { value: 30, label: '30 Cases' },
    { value: 50, label: '50 Cases' },
    { value: 100, label: '100 Cases' }
];

export const Preferences = ({ preferences, setPreferences, disabled, pageSizeOptions = PAGE_SIZE_OPTIONS }: any) => (
    <CollectionPreferences
        title="Preferences"
        confirmLabel="Confirm"
        cancelLabel="Cancel"
        disabled={disabled}
        preferences={preferences}
        onConfirm={({ detail }) => setPreferences(detail)}
        pageSizePreference={{
            title: 'Page size',
            options: pageSizeOptions
        }}
        wrapLinesPreference={{
            label: 'Wrap lines',
            description: 'Check to see all the text and wrap the lines'
        }}
        stripedRowsPreference={{
            label: 'Striped rows',
            description: 'Check to add alternating shaded rows'
        }}
        contentDensityPreference={{
            label: 'Compact mode',
            description: 'Check to display table content in a denser, more compact mode'
        }}
    />
);
