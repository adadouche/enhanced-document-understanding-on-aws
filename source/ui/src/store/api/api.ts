/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance      *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { BaseQueryFn, createApi, FetchArgs, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
import { generateToken, getRuntimeConfig } from '../../utils/apiHelper';

export const dynamicBaseQuery: BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError> = async (
    args: any,
    api: any,
    extraOptions: any
) => {
    const runtimeConfig = await getRuntimeConfig();
    const rawBaseQuery = fetchBaseQuery({
        baseUrl: runtimeConfig.ApiEndpoint,
        prepareHeaders: async (headers: any) => {
            if (process.env.NODE_ENV !== 'test') {
                const token = await generateToken();
                headers.set('Authorization', `${token}`);
            }
            return headers;
        }
    });
    return rawBaseQuery(args, api, extraOptions);
};
export const api = createApi({
    reducerPath: 'api',
    baseQuery: dynamicBaseQuery,
    tagTypes: ['Cases'],
    endpoints: () => ({
       
    }),
    refetchOnMountOrArgChange: true
});

