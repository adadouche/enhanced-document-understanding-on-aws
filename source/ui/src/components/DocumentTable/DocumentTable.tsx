/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import {
    Alert,
    AppLayout,
    Box,
    Button,
    Header,
    Pagination,
    SpaceBetween,
    SplitPanel,
    Table,
    TextContent,
    TextFilter
} from '@cloudscape-design/components';
import { API, Auth } from 'aws-amplify';
import { COLUMN_DEFINITIONS_MAIN, CasePreferences, DEFAULT_PREFERENCES } from './table-config';
import { EMPTY_PANEL_CONTENT, useSplitPanel } from './utils';

import { useCollection } from '@cloudscape-design/collection-hooks';
import React, { useCallback, useEffect } from 'react';
import { NavLink, useNavigate } from 'react-router-dom';
import { isStatusSuccess, renderStatus } from '../../utils/common-renderers';
import { API_NAME } from '../../utils/constants';
import { mapResultsToCases } from '../makeData';
import { TableEmptyState, TableNoMatchState } from './common-components';

import { CasePageHeader } from './full-page-header';
import { appLayoutAriaLabels } from './i18n-strings/app-layout';
import { paginationAriaLabels } from './i18n-strings/pagination';
import { splitPanelI18nStrings } from './i18n-strings/split-panel';
import { getTextFilterCounterText } from './i18n-strings/text-filter';
import { CaseTableInfoPanelContent } from '../../utils/info-panel-contents';

let token: string;
let results: any;

export interface DocumentTableProps {
    selectedDocumentId: string | null;
    setSelectedDocumentId: Function;
    selectedCaseId: string | null;
    setSelectedCaseId: Function;
    selectedDocumentFileType: string | null;
    setSelectedDocumentFileType: Function;
    setCasesList: Function;
    setSelectedCaseName: Function;
    numRequiredDocuments: number;
    requiredDocTypes?: string[];
    setSelectedDocumentName: Function;
}

export async function generateToken() {
    try {
        const user = await Auth.currentAuthenticatedUser();
        const token = user.getSignInUserSession().getIdToken().getJwtToken();
        return token;
    } catch (error) {
        console.error('error REST API:', error);
    }
}

export async function getUsername(): Promise<string | undefined> {
    try {
        const user = await Auth.currentAuthenticatedUser();
        const username = user.username;
        return username;
    } catch (error) {
        console.error('error REST API:', error);
    }
}

export default function DocumentTable(props: DocumentTableProps) {
    const [data, setData] = React.useState<any[]>([]);
    const navigate = useNavigate();
    const [currentStatus, setCurrentStatus] = React.useState('');
    const [toolsOpen, setToolsOpen] = React.useState(false);

    const fetchData = async () => {
        try {
            setCurrentStatus('loading');
            token = await generateToken();
            results = await API.get(API_NAME, 'cases', {
                headers: {
                    Authorization: token
                }
            });
            const newData = mapResultsToCases(results.Items);
            setData(newData);
            props.setCasesList(newData);
            setCurrentStatus('success');
        } catch (error) {
            setCurrentStatus('error');
            console.error(error);
        }
    };

    useEffect(() => {
        fetchData();
    }, []);

    const handleUploadButtonClick = useCallback((caseId: string) => {
        props.setSelectedCaseId(caseId);
        if (collectionProps.selectedItems?.length) {
            props.setSelectedCaseName(collectionProps.selectedItems[0].name);
        }
        navigate(`/uploadDocument`);
    }, []);

    const createDocumentLinkButton = (row: any, item: any) => {
        return (
            <Button variant="link" onClick={() => handleDocumentClick(item.caseId, row.docId, row.fileType, row.name)}>
                {row.name}
            </Button>
        );
    };

    const getPanelContent = (items: any) => {
        if (!items.length) {
            return EMPTY_PANEL_CONTENT;
        }
        const item = items[0];
        const columnDefinitions = [
            {
                id: 'name',
                header: 'Document Name',
                cell: (row: any) => createDocumentLinkButton(row, item)
            },
            {
                id: 'docId',
                header: 'Document ID',
                cell: (row: any) => row.docId
            },
            {
                id: 'dateCreated',
                header: 'Creation Date',
                cell: (row: any) => row.dateCreated
            },
            {
                id: 'fileType',
                header: 'File type',
                cell: (row: any) => row.fileType
            }
        ];

        return {
            header: item.name,
            body: (
                <Box>
                    <Table
                        header={
                            <Box>
                                <SpaceBetween direction="vertical" size="s">
                                    <Header
                                        variant="h2"
                                        counter={`(${item.caseDocuments.length})`}
                                        actions={
                                            <Box>
                                                <Button
                                                    variant="primary"
                                                    onClick={() => handleUploadButtonClick(item.caseId)}
                                                    data-testid="document-table-upload-document"
                                                    disabled={item.caseDocuments.length >= props.numRequiredDocuments}
                                                >
                                                    Upload Document
                                                </Button>
                                            </Box>
                                        }
                                    >
                                        Documents
                                    </Header>

                                    {item.caseDocuments.length >= props.numRequiredDocuments && (
                                        <Alert statusIconAriaLabel="Info" header="Required Documents Uploaded">
                                            Each case is configured to accept {props.numRequiredDocuments} document
                                            {props.numRequiredDocuments > 1 ? 's' : ''}.{' '}
                                            <NavLink to="/createCase">Create a new case</NavLink> to upload more
                                            documents
                                        </Alert>
                                    )}

                                    <TextContent>
                                        <p>
                                            Please upload <strong>{props.numRequiredDocuments}</strong> document
                                            {props.numRequiredDocuments > 1 ? 's' : ''} of type:{' '}
                                            <strong>{props.requiredDocTypes?.join(', ')}</strong> <br />
                                        </p>
                                    </TextContent>
                                </SpaceBetween>
                            </Box>
                        }
                        columnDefinitions={columnDefinitions}
                        items={item.caseDocuments}
                        variant="embedded"
                        data-testid="document-table-splitpanel-table"
                    />
                </Box>
            )
        };
    };

    const [preferences, setPreferences] = React.useState(DEFAULT_PREFERENCES);
    const { items, actions, filteredItemsCount, collectionProps, filterProps, paginationProps } = useCollection(data, {
        filtering: {
            empty: <TableEmptyState resourceName="Case" buttonOnClick={() => navigate(`/createCase`)} />,
            noMatch: <TableNoMatchState onClearFilter={() => actions.setFiltering('')} />
        },
        pagination: { pageSize: preferences.pageSize },
        selection: {}
        // sorting: {
        //     defaultState: {
        //         isDescending: descendingSorting,
        //         sortingColumn: sortingColumn
        //     }
        // }
    });

    useEffect(() => {
        if (collectionProps.selectedItems?.length) {
            props.setSelectedCaseId(collectionProps.selectedItems[0].caseId);
            props.setSelectedCaseName(collectionProps.selectedItems[0].name);
        }
    });

    const { header: panelHeader, body: panelBody } = getPanelContent(collectionProps.selectedItems);
    const { splitPanelOpen, onSplitPanelToggle, splitPanelSize, onSplitPanelResize } = useSplitPanel(
        collectionProps.selectedItems
    );

    const handleDocumentClick = useCallback(
        (caseId: string, docId: string, fileType: string, docName: string) => {
            const sanitizedFileType = fileType.replace('.', '');

            props.setSelectedDocumentId(docId);
            props.setSelectedCaseId(caseId);
            props.setSelectedDocumentFileType(sanitizedFileType);
            props.setSelectedDocumentName(docName);
            if (collectionProps.selectedItems?.length) {
                props.setSelectedCaseName(collectionProps.selectedItems[0].name);
                window.sessionStorage.setItem('selectedCaseName', collectionProps.selectedItems[0].name);
            }

            window.sessionStorage.setItem('selectedDocumentName', docName);
            window.sessionStorage.setItem('selectedDocumentId', docId);
            window.sessionStorage.setItem('selectedCaseId', caseId);
            window.sessionStorage.setItem('selectedDocumentFileType', sanitizedFileType);
            navigate(`${caseId}/document/${docId}`);
        },
        [collectionProps.selectedItems, navigate, props]
    );

    const tableContent = isStatusSuccess(currentStatus) ? (
        <Table
            {...collectionProps}
            header={
                <CasePageHeader
                    title="Cases"
                    createButtonText="Create case"
                    selectedItemsCount={collectionProps.selectedItems?.length || 0}
                    counter={'(' + data.length.toString() + ')'}
                    data-testid="document-table-header"
                    refreshFunction={fetchData}
                    onInfoLinkClick={() => {
                        setToolsOpen(true);
                    }}
                />
            }
            columnDefinitions={COLUMN_DEFINITIONS_MAIN}
            items={items}
            variant="full-page"
            stickyHeader={true}
            selectionType="single"
            filter={
                <TextFilter
                    {...filterProps}
                    filteringAriaLabel="Filter cases"
                    filteringPlaceholder="Find cases"
                    filteringClearAriaLabel="Clear"
                    countText={getTextFilterCounterText(filteredItemsCount || 0)}
                    data-testid="document-table-search"
                />
            }
            wrapLines={preferences.wrapLines}
            stripedRows={preferences.stripedRows}
            contentDensity={preferences.contentDensity as 'comfortable' | 'compact' | undefined}
            pagination={<Pagination {...paginationProps} ariaLabels={paginationAriaLabels} />}
            preferences={<CasePreferences preferences={preferences} setPreferences={setPreferences} />}
            data-testid="document-table-cases-table"
        />
    ) : (
        renderStatus(currentStatus, true, false, 'Error loading cases.', '')
    );

    return (
        <AppLayout
            contentType="table"
            headerSelector="#header"
            navigationHide
            toolsOpen={toolsOpen}
            tools={<CaseTableInfoPanelContent />}
            onToolsChange={({ detail }) => setToolsOpen(detail.open)}
            splitPanelOpen={splitPanelOpen}
            onSplitPanelToggle={onSplitPanelToggle}
            splitPanelSize={splitPanelSize}
            onSplitPanelResize={onSplitPanelResize}
            splitPanel={
                <SplitPanel
                    header={panelHeader}
                    i18nStrings={splitPanelI18nStrings}
                    data-testid="document-table-splitpanel"
                >
                    {panelBody}
                </SplitPanel>
            }
            content={tableContent}
            ariaLabels={appLayoutAriaLabels}
            data-testid="document-table-applayout"
        />
    );
}
