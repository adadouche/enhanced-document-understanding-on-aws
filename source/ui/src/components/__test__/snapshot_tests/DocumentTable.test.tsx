/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { API } from '@aws-amplify/api';
import { Auth } from '@aws-amplify/auth';
import DocumentTable from '../../DocumentTable/DocumentTable';

const mockAPI = {
    get: jest.fn(),
    post: jest.fn()
};
jest.mock('@aws-amplify/api');
API.get = mockAPI.get;
API.post = mockAPI.post;

const getCasesResponse = {
    Count: 4,
    ScannedCount: 4,
    Items: [
        {
            BUCKET_NAME: { S: 'fake-bucket-name' },
            CASE_ID: { S: 'fake-case-id' },
            CASE_NAME: { S: 'fake-case-name' },
            DOCUMENT_ID: { S: 'fake-doc-id' },
            DOCUMENT_TYPE: { S: 'passport' },
            S3_KEY: {
                S: 'fake-s3-key.jpg'
            },
            UPLOADED_FILE_EXTENSION: { S: 'jpg' },
            UPLOADED_FILE_NAME: { S: 'fake-file-name' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        }
    ]
};

let selectedDocumentId: string | null = null;
const setSelectedDocumentId = jest.fn();
let selectedCaseId: string | null = null;
const setSelectedCaseId = jest.fn();
let selectedDocumentFileType: string | null = null;
const setSelectedDocumentFileType = jest.fn();
let casesList: any[] | null = null;
const setCasesList = jest.fn();
const setSelectedCaseName = jest.fn();
const setSelectedDocumentName = jest.fn();
const documentTableProps = {
    selectedDocumentId: selectedDocumentId,
    setSelectedDocumentId: setSelectedDocumentId,
    selectedCaseId: selectedCaseId,
    setSelectedCaseId: setSelectedCaseId,
    selectedDocumentFileType: selectedDocumentFileType,
    setSelectedDocumentFileType: setSelectedDocumentFileType,
    casesList: casesList,
    setCasesList: setCasesList,
    setSelectedCaseName: setSelectedCaseName,
    numRequiredDocuments: 20,
    setSelectedDocumentName: setSelectedDocumentName
};

beforeEach(() => {
    mockAPI.get.mockReset();
    Auth.currentAuthenticatedUser = jest.fn().mockImplementation(() => {
        return {
            getSignInUserSession: jest.fn().mockImplementation(() => {
                return {
                    getIdToken: jest.fn().mockImplementation(() => {
                        return {
                            getJwtToken: jest.fn().mockImplementation(() => {
                                return 'fake-jwt-token';
                            })
                        };
                    })
                };
            })
        };
    });
});

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components') as any;
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

test('Snapshot test', async () => {
    mockAPI.get.mockResolvedValueOnce(getCasesResponse);
    const tree = renderer
        .create(
            <MemoryRouter initialEntries={['/']}>
                <Routes>
                    <Route path="/" element={<DocumentTable {...documentTableProps} />} />
                </Routes>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
