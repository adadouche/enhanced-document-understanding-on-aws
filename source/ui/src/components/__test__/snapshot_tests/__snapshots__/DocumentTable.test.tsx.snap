// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<AppLayout
  ariaLabels={
    Object {
      "navigation": "Side navigation",
      "navigationClose": "Close side navigation",
      "navigationToggle": "Open side navigation",
      "notifications": "Notifications",
      "tools": "Help panel",
      "toolsClose": "Close help panel",
      "toolsToggle": "Open help panel",
    }
  }
  content={null}
  contentType="table"
  data-testid="document-table-applayout"
  headerSelector="#header"
  navigationHide={true}
  onSplitPanelResize={[Function]}
  onSplitPanelToggle={[Function]}
  onToolsChange={[Function]}
  splitPanel={
    <SplitPanel
      data-testid="document-table-splitpanel"
      header="No case selected"
      i18nStrings={
        Object {
          "closeButtonAriaLabel": "Close panel",
          "openButtonAriaLabel": "Open panel",
          "preferencesCancel": "Cancel",
          "preferencesConfirm": "Confirm",
          "preferencesPositionBottom": "Bottom",
          "preferencesPositionDescription": "Choose the default split panel position for the service.",
          "preferencesPositionLabel": "Split panel position",
          "preferencesPositionSide": "Side",
          "preferencesTitle": "Split panel preferences",
          "resizeHandleAriaLabel": "Resize split panel",
        }
      }
    >
      Select a case to see its documents.
    </SplitPanel>
  }
  splitPanelOpen={false}
  splitPanelSize={500}
  tools={<CaseTableInfoPanelContent />}
  toolsOpen={false}
/>
`;
